docs/explanation/e-cryptography.md

```diff
@@ -0,0 +1,78 @@
+# Cryptography
+
+## Resource checksums
+
+Every version of the Charmed Kafka and Charmed ZooKeeper operators install a pinned revision of the Charmed Kafka snap
+and Charmed ZooKeeper, respectively, in order to 
+provide reproducible and secure environments. The Charmed Kafka snap and Charmed ZooKeeper snap package the 
+Kafka and ZooKeeper workload together with 
+a set of dependencies and utilities required by the lifecycle of the operators. 
+Every artifact bundled into the Charmed Kafka snap and Charmed ZooKeeper snap are verified against their checksum. 
+
+## Sources verification
+
+Charmed Kafka and Charmed ZooKeeper artifacts are published programmatically using release pipelines implemented via GitHub Actions of the 
+associated repositories. 
+All repositories containing the source code of the Charmed Kafka project (i.e. charm and snap codebases) are set up with 
+
+* Branch protection rules requiring additional commits to be merged via Pull-Request with at least 2 approvals from repository maintainers
+* Signed commits requirements (using GPG keys)
+* Requiring developers to sign CLA agreement
+
+## Encryption
+
+The Charmed Kafka operator allow to deploy a secure Kafka cluster providing encryption-in-transit capabilities out of the box 
+for
+* Interbroker communications
+* ZooKeeper connection
+* External client connection 
+
+In order to set up secure connection Kafka and ZooKeeper applications need to be integrated with TLS Certificate Provider charms, e.g. 
+`self-signed-certificates` operator. CSRs are generated for every unit using `tls_certificates_interface` library that uses `cryptography` 
+python library to create X.509 compatible certificates. The CSR is signed by the TLS Certificate Provider and returned to the units, and 
+stored in a password-protected Keystore file. The password of the Keystore is stored in Juju secrets starting from revision 168 on Kafka 
+and revision 130 on ZooKeeper. The relation provides also the certificate for the CA to be loaded in a password-protected Truststore file.
+
+When encryption is enabled, hostname verification is turned on for client connections, including inter-broker communication. 
+
+Encryption at rest is currently not supported, although it can be provided by the substrate (cloud or on-premises).
+
+## Authentication
+
+There exists 3 layers of authentication in the Charmed Kafka solution:
```

Comment:
```suggestion
There are three layers of authentication in the Charmed Kafka solution:
```

Are we sure we want to call them layers? I'd expect layers to be enveloping each other, like in an onion. I'm not sure it is the case here, but maybe it is, that's why I'm asking.

Reply:
done! I have also restructured the text, althouhg using the word layer for authentication is not that weird, since auth is indeed a layer that is pluggable on top of endpoints (with different protocols)